core/integration/git_test.go

```diff
@@ -116,6 +116,38 @@ func (GitSuite) TestGit(ctx context.Context, t *testctx.T) {
 	require.Contains(t, res.Git.Tag.Tree.File.Contents, "Dagger")
 }
 
+func (GitSuite) TestDiscardGitDir(ctx context.Context, t *testctx.T) {
+	c := connect(ctx, t)
+
+	t.Run("git dir is present", func(ctx context.Context, t *testctx.T) {
+		dir := c.Git("https://github.com/dagger/dagger").Branch("main").Tree()
+		ent, _ := dir.Entries(ctx)
```

Comment:
nit: always worth including the `require.NoError` check for these imo - having wasted time before chasing confounding second-order failures